src/volttron/services/control/control_service.py

```diff
@@ -410,9 +409,16 @@ def protocol_subscription(peer, sender, bus, topic, headers, message):
             response_received = True
 
         agent_uuid = self._raise_error_if_identity_exists_without_force(vip_identity, force)
+        if not agent.endswith(".whl"):
+            # agent passed is package name to install from pypi.
+            agent_uuid = self._aip.install_agent(agent, vip_identity, publickey, secretkey, agent_config, force)
```

Comment:
Nit. Could be just a one-liner by returning the output of the helper method.

Reply:
fixed